### PR TITLE
Look up correct property from request body object

### PR DESCRIPTION
**Pull request by AndrewFerr**
Also add some type-checking

## Checklist

* [ ] Tests written for all new code
* [x] Linter has been satisfied
* [x] Sign-off given on the changes (see CONTRIBUTING.md)

### CODE REVIEW

File: src/e2ee/RustEngine.ts [hunk 1]
```diff
@@ -107,7 +113,7 @@ export class RustEngine {
         settings.rotationPeriodMessages = BigInt(encEv.rotationPeriodMessages);
 
         await this.lock.acquire(roomId, async () => {
-            const requests = JSON.parse(await this.machine.shareRoomKey(new RoomId(roomId), members, settings));
+            const requests = JSON.parse(await this.machine.shareRoomKey(new RoomId(roomId), members, settings)) as RequestBody[];
```
Comment (turt2live) [inline src/e2ee/RustEngine.ts:116]: We should be getting types from the rust-sdk on this, largely to prevent this exact error

**Comment by AndrewFerr** [comment]
This PR indeed fixes matrix-org/matrix-rust-sdk#1093 and turt2live/matrix-bot-sdk#256.

Here's one way to trigger it:
1) create an encrypted room
2) invite a E2EE-capable bot to it
3) exchange some messages
4) invite any other user the room

Also, this seems to happen only when using bot-style / CS-API bots, not appservice-style bots, which makes sense since the problem happens during `/sync`.

File: src/e2ee/RustEngine.ts (hunk 1 above)
Comment (AndrewFerr) [reply src/e2ee/RustEngine.ts:116]: Does there exist a type for this yet?  The only mention of the request body properties that I could find was within a comment in the generated `index.js`/`index.d.ts`:
```
export class ToDeviceRequest {
  /** The request ID. */
  readonly id: string
  /**
   * A JSON-encoded object of form:
   *
   * ```json
   * {"event_type": …, "txn_id": …, "messages": …}
   * ```
   */
  readonly body: string
  /** Get its request type. */
  get type(): RequestType
}
```

Or is the fact that no TS type is generated for `body` what you meant by ["the TS types for the rust-sdk are likely wrong"](https://github.com/matrix-org/matrix-rust-sdk/issues/1093#issuecomment-1296957629)?

**Comment by turt2live** [comment]
I've opened https://github.com/turt2live/matrix-bot-sdk/pull/288 to cover the bug while we wait for the rust-sdk to have better types.

**Pull request closed without merging**